### PR TITLE
Added mod_actions reason aliases

### PR DESCRIPTION
**Pull request by metal0**

### CODE REVIEW

File: backend/src/plugins/ModActions/functions/parseReason.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const MAX_REASON_LENGTH = 512;
+
+export function parseReason(config: any, reason: string): string {
+  if (!reason) return reason;
+  if (config?.reason_aliases) {
+    reason = config.reason_aliases![reason.toLowerCase()] ?? reason;
+  }
+  if (reason!.length > MAX_REASON_LENGTH) {
+    reason = reason!.substring(0, MAX_REASON_LENGTH - 6) + " [...]";
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/parseReason.ts:9]:
```suggestion
    reason = reason!.substring(0, MAX_REASON_LENGTH - 4) + " […]";
```

File: backend/src/plugins/ModActions/commands/UnbanCmd.ts [hunk 2]
```diff
@@ -48,7 +48,8 @@ export const UnbanCmd = modActionsCmd({
     }
 
     pluginData.state.serverLogs.ignoreLog(LogType.MEMBER_UNBAN, user.id);
-    const reason = formatReasonWithAttachments(args.reason, [...msg.attachments.values()]);
+    const config = pluginData.config.get();
+    const reason = parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]));
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/UnbanCmd.ts:52]:
```suggestion
    const reason = formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()]);
```

File: backend/src/plugins/ModActions/functions/parseReason.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const MAX_REASON_LENGTH = 512;
+
+export function parseReason(config: any, reason: string): string {
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/parseReason.ts:3]: There has to be a way to type the `config` argument here

File: backend/src/plugins/ModActions/functions/parseReason.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  if (!reason) return reason;
+  if (config?.reason_aliases) {
+    reason = config.reason_aliases![reason.toLowerCase()] ?? reason;
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/parseReason.ts:6]: I can't imagine the `!` being needed here
```suggestion
    reason = config.reason_aliases[reason.toLowerCase()] ?? reason;
```

Comment (almeidx) [inline backend/src/plugins/ModActions/functions/parseReason.ts:6]: If you're making the `reason` lowercase here, you should make the `reason_aliases` keys lowercase too

File: backend/src/plugins/ModActions/functions/actualUnmuteUserCmd.ts [hunk 5]
```diff
@@ -27,7 +28,10 @@ export async function actualUnmuteCmd(
     pp = msg.author;
   }
 
-  const reason = args.reason ? formatReasonWithAttachments(args.reason, [...msg.attachments.values()]) : undefined;
+  const config = pluginData.config.get();
+  const reason = args.reason
+    ? parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]))
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/actualUnmuteUserCmd.ts:33]:
```suggestion
    ? formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()])
```

File: backend/src/plugins/ModActions/functions/actualMuteUserCmd.ts [hunk 6]
```diff
@@ -42,7 +43,10 @@ export async function actualMuteUserCmd(
   }
 
   const timeUntilUnmute = args.time && humanizeDuration(args.time);
-  const reason = args.reason ? formatReasonWithAttachments(args.reason, [...msg.attachments.values()]) : undefined;
+  const config = pluginData.config.get();
+  const reason = args.reason
+    ? parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]))
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/actualMuteUserCmd.ts:48]:
```suggestion
    ? formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()])
```

File: backend/src/plugins/ModActions/functions/actualKickMemberCmd.ts [hunk 7]
```diff
@@ -67,7 +68,8 @@ export async function actualKickMemberCmd(
     return;
   }
 
-  const reason = formatReasonWithAttachments(args.reason, msg.attachments);
+  const config = pluginData.config.get();
+  const reason = parseReason(config, formatReasonWithAttachments(args.reason, msg.attachments));
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/actualKickMemberCmd.ts:72]: Also, if `msg` is a Message from discord.js, this won't work. (`msg` is typed as any so I don't really know)

```suggestion
  const reason = formatReasonWithAttachments(parseReason(config, args.reason), msg.attachments);
```

File: backend/src/plugins/ModActions/commands/WarnCmd.ts [hunk 8]
```diff
@@ -63,7 +64,7 @@ export const WarnCmd = modActionsCmd({
     }
 
     const config = pluginData.config.get();
-    const reason = formatReasonWithAttachments(args.reason, [...msg.attachments.values()]);
+    const reason = parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]));
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/WarnCmd.ts:67]:
```suggestion
    const reason = formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()]);
```

File: backend/src/plugins/ModActions/commands/MassmuteCmd.ts [hunk 9]
```diff
@@ -40,7 +40,11 @@ export const MassmuteCmd = modActionsCmd({
       return;
     }
 
-    const muteReason = formatReasonWithAttachments(muteReasonReceived.content, [...msg.attachments.values()]);
+    const config = pluginData.config.get();
+    const muteReason = parseReason(
+      config,
+      formatReasonWithAttachments(muteReasonReceived.content, [...msg.attachments.values()]),
+    );
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/MassmuteCmd.ts:47]:
```suggestion
    const muteReason = formatReasonWithAttachments(
      parseReason(config, muteReasonReceived.content),
      [...msg.attachments.values()],
    );
```

File: backend/src/plugins/ModActions/commands/MassUnbanCmd.ts [hunk 10]
```diff
@@ -37,8 +37,11 @@ export const MassunbanCmd = modActionsCmd({
       sendErrorMessage(pluginData, msg.channel, "Cancelled");
       return;
     }
-
-    const unbanReason = formatReasonWithAttachments(unbanReasonReply.content, [...msg.attachments.values()]);
+    const config = pluginData.config.get();
+    const unbanReason = parseReason(
+      config,
+      formatReasonWithAttachments(unbanReasonReply.content, [...msg.attachments.values()]),
+    );
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/MassUnbanCmd.ts:44]:
```suggestion
    const unbanReason = formatReasonWithAttachments(
      parseReason(config, unbanReasonReply.content),
      [...msg.attachments.values()],
    );
```

File: backend/src/plugins/ModActions/commands/ForcebanCmd.ts [hunk 11]
```diff
@@ -61,8 +62,8 @@ export const ForcebanCmd = modActionsCmd({
 
       mod = args.mod;
     }
-
-    const reason = formatReasonWithAttachments(args.reason, [...msg.attachments.values()]);
+    const config = pluginData.config.get();
+    const reason = parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]));
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/ForcebanCmd.ts:66]:
```suggestion
    const reason = formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()]);
```

File: backend/src/plugins/ModActions/commands/AddCaseCmd.ts [hunk 12]
```diff
@@ -60,8 +59,8 @@ export const AddCaseCmd = modActionsCmd({
       sendErrorMessage(pluginData, msg.channel, "Cannot add case: invalid case type");
       return;
     }
-
-    const reason = formatReasonWithAttachments(args.reason, [...msg.attachments.values()]);
+    const config = pluginData.config.get();
+    const reason = parseReason(config, formatReasonWithAttachments(args.reason, [...msg.attachments.values()]));
```
Comment (almeidx) [inline backend/src/plugins/ModActions/commands/AddCaseCmd.ts:63]:
```suggestion
    const reason = formatReasonWithAttachments(parseReason(config, args.reason), [...msg.attachments.values()]);
```

File: backend/src/plugins/ModActions/functions/banUserId.ts [hunk 13]
```diff
@@ -38,6 +39,7 @@ export async function banUserId(
       error: "Invalid user",
     };
   }
+  if (reason) reason = parseReason(config, reason);
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/banUserId.ts:42]:
```suggestion
  reason &&= parseReason(config, reason);
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — CHANGES_REQUESTED** [review]

File: backend/src/plugins/ModActions/functions/actualKickMemberCmd.ts [hunk 14]
```diff
@@ -1,6 +1,7 @@
-import { GuildMember, TextChannel, ThreadChannel } from "discord.js";
+import { GuildMember, Message, TextChannel, ThreadChannel } from "discord.js";
 import { GuildPluginData } from "knub";
 import { hasPermission } from "knub/dist/helpers";
+import { GuildMessage } from "knub/dist/types";
```
Comment (almeidx) [inline backend/src/plugins/ModActions/functions/actualKickMemberCmd.ts:4]:
```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by isaacKenyon** [comment]
> ```yaml
> plugins:
>   mod_actions:
>    reason_aliases:
>      advertising: This is an alias for advertising
> ```
> 
> > `!ban @user advertising`
> 
> The `reason_aliases` keys are not case sensitive

`reason_aliases` should be nested under the `config:`

**Comment by Ibotmealot** [comment]
Dunno if this isn't intended, however, when using the update command and if you use an aliases it doesn't display the aliases. Hope that makes sense?

**Comment by metal0** [comment]
Superded by #390

**Pull request closed without merging**